### PR TITLE
[NT-1990] Feature Flag cleanup

### PR DESCRIPTION
**Pull request by singhhari**
# 📲 What

There are a handful of feature flags that can be safely deprecated on our project. That includes flags associated with 'Sign In With Apple', Kickstarters 'Lights On' promotion, and 'Email Verification`. These features are either live in production or removed from future consideration.

# 🤔 Why

We are metaphorically pruning the tree, which is our codebase. If feature flags are not removed after they have served their purpose they contribute to technical debt and bloat.  

# 🛠 How

We have two files with features on them -- `OptimizelyFeature.swift` and `Feature.swift`. Both of these files serve the same role but provide flags from different sources. Removing the flags and their associated code involves us going into view models, view controllers and test files to ensure we are completely removing all references.

# ✅ Acceptance criteria

- [ ] Verify Sign In With Apple works without regression.
- [ ] Verify the code removal of `Lights On` did not introduce regressions in the Discovery functionality.

### CODE REVIEW

File: Library/ViewModels/DiscoveryPageViewModel.swift [hunk 1]
```diff
@@ -359,24 +350,6 @@ public final class DiscoveryPageViewModel: DiscoveryPageViewModelType, Discovery
         return sort == .magic && filterParams == DiscoveryViewModel.initialParams()
       }
 
-    let featureFlagValue = self.sortProperty.signal.skipNil()
```
Comment (singhhari) [inline Library/ViewModels/DiscoveryPageViewModel.swift:362]: I'm not certain if the code under `// MARK: - Editorial Header` is safe to deprecate as well?

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by singhhari** [comment]
Coverage down quite a bit here. I suspect that is expected given the removal of all this code?

**Review by msadoon — APPROVED** [review]
Codewise, everything looks fine, code-coverage only went down slightly. 

I can't test Apple Sign In on the simulator, it hangs for some reason (even on `master`, so it's not this PR) -- ordered a cable to use my personal iOS device for testing in the future.

If anyone got past this please let me know.

To test the `lights_on` experiment on Optimizely dashboard I checked Everyone Audience set to `100%` on Staging on `master` -- was showing the `DiscoveryEditorialCell`, and on this same setting for this branch was not showing anything - which tells me the feature is gone and won't be affected by the flag 👍🏿

File: Library/ViewModels/LoginToutViewModel.swift [hunk 2]
```diff
@@ -53,9 +53,6 @@ public protocol LoginToutViewModelInputs {
 }
 
 public protocol LoginToutViewModelOutputs {
```
[the remark this answers is not in the document]
Comment (singhhari) [reply Library/ViewModels/LoginToutViewModel.swift:55]: Ah so the definition of "tout" is to solicit so I interpreted this as us "touting" the user to login.

**Comment by singhhari** [comment]
> Codewise, everything looks fine, code-coverage only went down slightly.
> 
> I can't test Apple Sign In on the simulator, it hangs for some reason (even on `master`, so it's not this PR) -- ordered a cable to use my personal iOS device for testing in the future.
> 
> If anyone got past this please let me know.
> 
> To test the `lights_on` experiment on Optimizely dashboard I checked Everyone Audience set to `100%` on Staging on `master` -- was showing the `DiscoveryEditorialCell`, and on this same setting for this branch was not showing anything - which tells me the feature is gone and won't be affected by the flag 👍🏿

Yep this is an issue with Sign In With Apple on the simulator. I think it has something to do with setting up your Apple ID and it has historically been difficult to test this sort of functionality. I believe when the feature was developed testing this was quite painful.

File: Library/ViewModels/DiscoveryPostcardViewModelTests.swift [hunk 3]
```diff
@@ -398,28 +398,24 @@ internal final class DiscoveryPostcardViewModelTests: TestCase {
     self.projectCategoryViewHidden.assertValue(true)
   }
 
-  func testLocationStackView_IsNotHidden_WhenTagIsLightsOn() {
+  func testLocationStackView_IsHidden() {
```
[the remark this answers is not in the document]
Comment (singhhari) [reply Library/ViewModels/DiscoveryPostcardViewModelTests.swift]: Updated.

File: Library/ViewModels/DiscoveryPostcardViewModel.swift [hunk 4]
```diff
@@ -276,9 +276,8 @@ public final class DiscoveryPostcardViewModel: DiscoveryPostcardViewModelType,
     let params = self.configureWithValueProperty.signal.skipNil().map(third)
 
     self.locationStackViewHidden = params.map { params in
-      guard let params = params, let tagId = params.tagId else { return true }
-
-      return tagId != DiscoveryParams.TagID.lightsOn
+      guard let _ = params else { return false }
+      return true
```
[the remark this answers is not in the document]
Comment (singhhari) [reply Library/ViewModels/DiscoveryPostcardViewModel.swift]: Updated.

File: Library/ViewModels/DiscoveryPageViewModel.swift (hunk 1 above)
Comment (justinswart) [reply Library/ViewModels/DiscoveryPageViewModel.swift:362]: Yeah I think that can go.

**Comment by justinswart** [comment]
I think all of these can also be removed:
image

**Review by justinswart — APPROVED** [review]

**Review by moderateepheezy — APPROVED** [review]

**Pull request merged**